cpp/arcticdb/storage/azure/azure_client_impl.cpp

```diff
@@ -5,15 +5,46 @@
  * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
  */
 
+/**
+ * Azure Transport Selection:
+ * 
+ * This file implements platform-specific HTTP transport selection for Azure storage operations:
+ * 
+ * - On Windows (_WIN32):
+ *   - Always uses WinHTTP (native Windows HTTP client)
+ *   - CA certificate settings are not supported as WinHTTP uses Windows certificate store
+ *   - Custom CA certificates must be installed via Windows Certificate Manager (certmgr.msc)
+ *   - This provides better performance and integration with Windows security
+ * 
+ * - On macOS (__APPLE__):
+ *   - Always uses libcurl as the HTTP transport
+ *   - CA certificate settings are not supported
+ *   - Custom CA certificates must be installed via Keychain Access
+ * 
+ * - On Linux:
+ *   - Always uses libcurl as the HTTP transport
+ *   - Supports custom CA certificate configuration via ca_cert_path and ca_cert_dir
```

Comment:
I think we need to add notes about the Linux and macOS builds on conda-forge since they use conda-forge's distribution of `libcurl` which uses specific CA certificates.

Reply:
Done with dbdb38a.